### PR TITLE
Clean github functions

### PR DESCRIPTION
**Pull request by ronplaymeta**
Hi @pablo-mayrgundter 
I updated GitHub related functions for better understanding and performance.
Please review and give me feedback.

### CODE REVIEW

File: src/utils/GitHub.js
Comment (pablo-mayrgundter) [inline src/utils/GitHub.js:1]: This basically looks good.  So you removed a bunch of duplication for setting the auth headers.. and now it's all further down file?

And a new delete method?  Please add a test for that.

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]
Leaving for Ogali to approve.. but looks fine to me

**Comment by OlegMoshkovich** [comment]
Hi, @ronplaymeta I merged work that had failing tests on Friday, since then I reverted the commit.
Can you please rebase to the latest main?
Thank you.

**Comment by ronplaymeta** [comment]
Hi @OlegMoshkovich, thank you for letting me know. :)

File: src/utils/GitHub.js
Comment (ronplaymeta) [reply src/utils/GitHub.js:1]: Yep, I removed authentication duplication for better understanding.
Currently, only added github features that we're using now, but I think we have enough funcs for our project.
Thank you.

Comment (ronplaymeta) [reply src/utils/GitHub.js:1]: Ah, yep, i added new delete method, will add unit tests soon.

File: src/__mocks__/api-handlers.js
[the remark this answers is not in the document]
Comment (ronplaymeta) [reply src/__mocks__/api-handlers.js:1]: I reverted it.

File: src/__mocks__/api-handlers.js [hunk 1]
```diff
@@ -121,11 +116,19 @@ export const handlers = [
     )
   }),
 
+  rest.post('https://api.github.com/repos/:org/:repo/issues/:issueNumber/comments', (req, res, ctx) => {
```
[the remark this answers is not in the document]
Comment (ronplaymeta) [reply src/__mocks__/api-handlers.js:131]: issue number checking is added. :)

**Comment by ronplaymeta** [comment]
ptal

**Review by pablo-mayrgundter — COMMENTED** [review]
LGTM.  Leaving to Ogali for approval

File: src/__mocks__/api-handlers.js [hunk 2]
```diff
@@ -66,36 +71,36 @@ export const handlers = [
           download_url: downloadURL,
           type: 'file',
           content: 'U2hhcmUgaXMgYSB3ZWItYmFzZWQgQklNICYgQ0FEIGludGVncmF0aW9uIGVu\n' +
-              'dmlyb25tZW50IGZyb20gW2JsZHJzLmFpXShodHRwczovL2JsZHJzLmFpLyku\n' +
```
Comment (oo-bldrs) [inline src/__mocks__/api-handlers.js:69]: There's still this large block of whitespace changes -- can you revert this?

**Comment by ronplaymeta** [comment]
ptal

File: src/__mocks__/api-handlers.js (hunk 2 above)
Comment (ronplaymeta) [reply src/__mocks__/api-handlers.js]: done

**Review by oo-bldrs — APPROVED** [review]

**Review by pablo-mayrgundter — APPROVED** [review]

**Pull request merged**